core/logger/default.go

```diff
@@ -38,18 +38,18 @@ func init() {
 // Instead, you should fork the logger.Default instance to create a new logger
 // for your module.
 // Eg: logger.Default.Named("<my-package-name>")
-func SetLogger(newLogger *Logger) {
+func SetLogger(newLogger Logger) {
 	if Default != nil {
-		defer func() {
-			if err := Default.Sync(); err != nil {
+		defer func(l Logger) {
+			if err := l.Sync(); err != nil {
 				if errors.Unwrap(err).Error() != os.ErrInvalid.Error() &&
 					errors.Unwrap(err).Error() != "inappropriate ioctl for device" &&
 					errors.Unwrap(err).Error() != "bad file descriptor" {
 					// logger.Sync() will return 'invalid argument' error when closing file
 					log.Fatalf("failed to sync logger %+v", err)
 				}
 			}
-		}()
+		}(Default)
```

Comment:
Previously the value of the `Default` variable was not captured, so the deferred func would call `newLogger.Sync()`.